lib/core/src/lib/datatable/components/datatable/datatable.component.ts

```diff
@@ -407,6 +405,10 @@ export class DataTableComponent implements OnInit, AfterContentInit, OnChanges,
         return column.key === this.data.getSorting().key;
     }
 
+    getVisibleColumns(): DataColumn[] {
+        return this.data.getColumns().filter((column) => !column.isHidden);
+    }
+
```

Comment:
we can also reuse this function [here](https://github.com/Alfresco/alfresco-ng2-components/blob/1fb5a85426395bae4256e19933b14e0f64e19147/lib/core/src/lib/datatable/components/datatable/datatable.component.ts#L987)

Reply:
good finding, thanks!